File: src/main/java/com/tobe/healthy/lessonhistory/repository/LessonHistoryCommentRepositoryCustom.java
package com.tobe.healthy.lessonhistory.repository;

import org.springframework.stereotype.Repository;

import com.tobe.healthy.lessonhistory.domain.entity.LessonHistoryComment;

@Repository
public interface LessonHistoryCommentRepositoryCustom {
	int findTopComment(Long lessonHistoryId, Long lessonHistoryCommentId);

	LessonHistoryComment findLessonHistoryCommentWithFiles(Long lessonHistoryCommentId, Long writerId);

	LessonHistoryComment findCommentById(Long lessonHistoryCommentId);

	LessonHistoryComment findById(Long lessonHistoryCommentId, Long writerId);
}

File: src/main/kotlin/com/tobe/healthy/lessonhistory/repository/LessonHistoryCommentRepositoryCustom.kt
package com.tobe.healthy.lessonhistory.repository

import com.tobe.healthy.lessonhistory.domain.entity.LessonHistoryComment
import org.springframework.stereotype.Repository

@Repository
interface LessonHistoryCommentRepositoryCustom {
    fun findTopComment(lessonHistoryId: Long?): Int
    fun findTopComment(lessonHistoryId: Long?, lessonHistoryCommentId: Long?): Int
    fun findLessonHistoryCommentWithFiles(lessonHistoryCommentId: Long): LessonHistoryComment?
    fun findCommentById(lessonHistoryCommentId: Long): LessonHistoryComment?
}
